flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/SpillingAdaptiveSpanningRecordDeserializer.java

```diff
@@ -557,6 +579,26 @@ private void addNextChunkFromMemorySegment(MemorySegment segment, int offset, in
 			}
 		}
 
+		@Nullable
+		MemorySegment copyToTargetSegment() {
+			// for the case of only partial length, no data
+			if (lengthBuffer.position() > 0) {
+				MemorySegment segment = MemorySegmentFactory.allocateUnpooledSegment(lengthBuffer.remaining());
```

Comment:
Could that fail for some reason? (maximum back pressure scenario)

Reply:
We allocate an overhead segment here for copying the data, so it is no related to back pressure scenario, only some memory overhead. Here I guess we might not use the framework managed memory.